controller/source/query.cpp

```diff
@@ -64,10 +56,12 @@ query::query(const std::string &input)
         this->m_expiration = stoi(value);
       }
       else if (option == "-objPur") {
-        set_bitmap(this->m_purpose, split_comma_string(value));
+        this->m_purpose.emplace();
+        set_bitmap(this->m_purpose.value(), split_comma_string(value));
       }
       else if (option == "-objObjections") {
-        set_bitmap(this->m_objection, split_comma_string(value));
+        this->m_objection.emplace();
```

Comment:
why do you need this? it was not clear to me from the commit message

Reply:
I changed the query metadata fields to `optional`. Since we are passing the underlying value to `set_bitmap` method, I think the value should be initialized. And I am doing this with `emplace` call before passing it to the function.